library/classes/Document.class.php

```diff
@@ -283,6 +283,16 @@ public function has_expired()
         return false;
     }
 
+    public function can_patient_access($pid)
```

Comment:
@bradymiller If you can look at this from a security point of view and let me know if you see any problems.  Its called in the FhirDocumentRestController.

Reply:
would just make sure $pid is also not empty in the if (overkill but might as well). otherwise looks good.

Reply:
@bradymiller Maybe I'm tired but I believe the first check for empty on the foreignID makes sure the $pid is also never empty since foreignID has to equal the $pid.

Reply:
that is true. is overkill, on security related stuff, I overuse !empty() and am explicit so don't need to think  :)
can leave as is.